deploy_pulp3/pulp3/tasks/main.yml

```diff
@@ -0,0 +1,85 @@
+---
+# tasks file for pulp3
+- name: Create user {{ pulp_user }}
+  user:
+    name: "{{ pulp_user }}"
+    comment: "Pulp user"
+
+- name: Get user's home directory
+  shell: 'getent passwd "{{ pulp_user }}" | cut -d: -f6'
+  check_mode: false
+  changed_when: false
+  register: result
+
+- name: Set variable referencing user's home directory
+  set_fact:
+    pulp_user_home: "{{ result.stdout }}"
+
+- name: Make user's home directory world-readable
+  file:
+    path: "{{ pulp_user_home }}"
+    state: directory
+    mode: 0755
+
+- name: Create ~/.bashrc.d/
+  file:
+    path: "{{ pulp_user_home }}/.bashrc.d/"
+    state: directory
+  become_user: "{{ pulp_user }}"
+
+- name: Create directory /var/lib/pulp
+  file:
+    path: /var/lib/pulp
+    state: directory
+    owner: "{{ pulp_user }}"
+    group: "{{ pulp_user }}"
+    mode: 0775
+
+- name: Install pulpcore
+  pip:
+    name: pulpcore
+    state: latest
+    virtualenv: "{{ pulp_user_home }}/pulpvenv"
+    virtualenv_python: python3
+  become_user: "{{ pulp_user }}"
+
+- name: Add Django and Pulp supplemental bashrc
+  copy:
+    src: files/supplem.bashrc
```

Comment:
Even if you just specify the filename, Ansible will pick it up automatically from the `files` directory of the role.

Reply:
I find it very useful to always specify files relative to the `files` directory. That lets me write code like this:

```yaml
- name: Install systemd units
  copy:
    src: etc_systemd_system/
    dest: /etc/systemd/system/
    owner: root
    group: root
    mode: 0644
```

In this example, all the files in `files/etc_systemd_system/` are copied into the target directory.